### PR TITLE
Add NativeLogger for android

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

Implements a native logger on android

### CODE REVIEW

File: shared/util/string.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// @flow
+
+function toStringForLog(a: any): string {
+  if (typeof a === 'string') {
+    return a
+  } else if (a instanceof Error) {
+    return a.toString()
```
Comment (akalin-keybase) [inline shared/util/string.js]: `Error.toString()` doesn't include the error stack, which is probably useful information. I suggest building a string out of `a.message` and/or `a.stack` instead

File: shared/util/string.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  } else if (typeof a === 'object') {
+    return JSON.stringify(a)
+  }
+  return (a.toString && a.toString()) || 'Failed to turn item to string in toStringForLog'
```
Comment (akalin-keybase) [inline shared/util/string.js]: it seems this should go before the object check, right? otherwise things with `toString` will be `JSON.stringify`ed first.

Comment (MarcoPolo) [reply shared/util/string.js]: every object has a .toString which returns `[object object]`

Comment (akalin-keybase) [reply shared/util/string.js]: d'oh, right

**Pull request merged**